state.go

```diff
@@ -177,6 +194,13 @@ func (a *Autopilot) nextState(ctx context.Context) (*State, error) {
 	return state, nil
 }
 
+// ComputeState will compute a new state via the normal means but will not retain it
+// internally. This functions main purpose is to help with testing promoter and
+// application integration interface implementations.
+func (a *Autopilot) ComputeState(ctx context.Context) (*State, error) {
```

Comment:
This method was added to facilitate testing of test computation when integrated with implementations of the `ApplicationIntegration` and `Promoter` interfaces when those were defined outside of this package.